diff -ruNa appweb-7.0.1-ori/src/http/httpLib.c appweb-7.0.1/src/http/httpLib.c
--- appweb-7.0.1-ori/src/http/httpLib.c	2017-09-28 11:36:09.000000000 +0800
+++ appweb-7.0.1/src/http/httpLib.c	2021-10-04 07:47:52.110535937 +0800
@@ -17327,7 +17327,13 @@
                 httpBadRequestError(conn, HTTP_CODE_BAD_REQUEST, "Bad host");
             }
         } else {
-            up->port = conn->sock->listenSock->port;
+            char *p = schr(hostname, ':');
+            if (p) {
+                up->port = atoi(p + 1);
+            }
+            else {
+                up->port = conn->sock->listenSock->port;
+            }
         }
     }
 }
