diff -ruNa appweb-7.0.1-ori/paks/osdep/dist/osdep.h appweb-7.0.1/paks/osdep/dist/osdep.h
--- appweb-7.0.1-ori/paks/osdep/dist/osdep.h	2017-09-28 11:36:09.000000000 +0800
+++ appweb-7.0.1/paks/osdep/dist/osdep.h	2026-07-03 16:34:43.051077446 +0800
@@ -56,7 +56,7 @@
     #define ME_CPU_ARCH ME_CPU_ALPHA
     #define CPU_ENDIAN ME_LITTLE_ENDIAN
 
-#elif defined(__arm__)
+#elif defined(__arm__) || defined(__aarch64__)
     #define ME_CPU "arm"
     #define ME_CPU_ARCH ME_CPU_ARM
     #define CPU_ENDIAN ME_LITTLE_ENDIAN
diff -ruNa appweb-7.0.1-ori/src/http/httpLib.c appweb-7.0.1/src/http/httpLib.c
--- appweb-7.0.1-ori/src/http/httpLib.c	2017-09-28 11:36:09.000000000 +0800
+++ appweb-7.0.1/src/http/httpLib.c	2026-07-03 16:34:18.858547042 +0800
@@ -17327,7 +17327,13 @@
                 httpBadRequestError(conn, HTTP_CODE_BAD_REQUEST, "Bad host");
             }
         } else {
-            up->port = conn->sock->listenSock->port;
+            char *p = schr(hostname, ':');
+            if (p) {
+                up->port = atoi(p + 1);
+            }
+            else {
+                up->port = conn->sock->listenSock->port;
+            }
         }
     }
 }
diff -ruNa appweb-7.0.1-ori/src/osdep/osdep.h appweb-7.0.1/src/osdep/osdep.h
--- appweb-7.0.1-ori/src/osdep/osdep.h	2017-09-28 11:36:09.000000000 +0800
+++ appweb-7.0.1/src/osdep/osdep.h	2026-07-03 16:34:38.814984565 +0800
@@ -56,7 +56,7 @@
     #define ME_CPU_ARCH ME_CPU_ALPHA
     #define CPU_ENDIAN ME_LITTLE_ENDIAN
 
-#elif defined(__arm__)
+#elif defined(__arm__) || defined(__aarch64__)
     #define ME_CPU "arm"
     #define ME_CPU_ARCH ME_CPU_ARM
     #define CPU_ENDIAN ME_LITTLE_ENDIAN
